### PR TITLE
[WFLY-19464] [Multiple -with-tools BOMs] Aligns user BOMs with server dists

### PR DESCRIPTION
**Pull request by emmartins**
**Issue:** https://issues.redhat.com/browse/WFLY-19464

This PR moves all server oriented BOMs to /boms/user/server, renames microprofile BOM to expansion, introduces preview variants of the ee and expansion BOMs, and provides with-tools BOM for each of those 4 BOMs

**Full Change log:**
* Introduced /server to hold all server usage BOM Builders (includes common to all BOMs dependency management and plugin config)
* Introduced /server/ee to hold all server usage EE BOM Builders
* Introduced /server/expansion to hold all server usage Expansion BOM Builders
* Moved EE BOM builder (/ee) to /server/ee/ee-standard
* Moved EE With Tools BOM builder (/ee/ee-with-tools) to /server/ee/ee-standard/ee-standard-with-tools
* Moved MicroProfile BOM builder (/microprofile) to /server/expansion/expansion-standard, and MicroProfile BOM renamed to Expansion BOM (coordinates org.wildfly.bom:microprofile—> org.wildfly.bom:expansion)
* Moved all builders pom.xml groupId, from org.wildfly.bom to org.wildfly.bom.builder (so new parents and builders doesn’t clash with existent built BOMs coordinates, and/or confuse users) 
* EE BOM common dependency management and plugin config moved from /server/ee/standard to /server/ee
* Expansion (ex-MicroProfile) BOM common dependency management and plugin config moved from /server/expansion/standard to /server/expansion
* Expansion BOM’s parent moved to EE BOM (thus inheriting all its dependency management) 
* Introduced EE Preview BOM Builder at /server/ee/ee-preview (artifacts included are the ones exclusive to EE Preview distribution, plus the inherited/append ones from parents).
* Introduced Expansion Preview BOM Builder at /server/expansion/expansion-preview (artifacts included are the ones exclusive to Expansion Preview distribution, plus the inherited/append ones from parents). Please note that Expansion Preview BOM’s parent is EE Preview BOM (thus inheriting all its dependency management too)
* Introduced builder for EE Preview With Tools BOM at /server/ee/ee-preview/ee-preview-with-tools
* Introduced builder for Expansion Standard With Tools BOM at /server/expansion/expansion-standard/expansion-standard-with-tools
* Introduced builder for Expansion Preview With Tools BOM at /server/expansion/expansion-preview/expansion-preview-with-tools

### CODE REVIEW

**Comment by emmartins** [comment]
@bstansberry here is the BOMs "rework" PoC, in this particular case we will deliver a -with-tools BOM for each "server" BOM (ee, ee-preview, expansion, expansion-preview)

**Comment by bstansberry** [comment]
Output comparisons vs main:

EE - https://gist.github.com/bstansberry/edf65f54a6f1e0f3e483f19d9ad57f2c

Expansion - https://gist.github.com/bstansberry/aa603c80f8ee6a24d07fef63ba45b091

**Comment by bstansberry** [comment]
Duplicating the tools stuff four times certainly illustrates the advantage of the 'Single Tools BOM' approach.

I think we'd really regret forcing a migration step on users though. People really hate migration. :(

File: boms/user/pom.xml [hunk 1]
```diff
@@ -34,12 +34,19 @@
...
     <artifactId>wildfly</artifactId>
 
     <packaging>pom</packaging>
 
-    <name>WildFly BOMs: Parent Aggregator</name>
+    <name>WildFly BOM Builders</name>
+
+    <properties>
+        <!-- the BOMs built's groupId -->
+        <bomGroupId>org.wildfly</bomGroupId>
```
Comment (bstansberry) [inline boms/user/pom.xml:46]: It seems we've been inconsistent about the bom groupId between the client and server boms, with the latter using org.wildfly.bom and the clients just org.wildfly. This PR switches them all to org.wildfly.

File: boms/user/server/ee/ee-preview/ee-preview-with-tools/pom.xml [hunk 2]
```diff
@@ -0,0 +1,168 @@
...
+                                <groupId>${bomGroupId}</groupId>
+                                <artifactId>${bomArtifactIdPrefix}-ee-preview</artifactId>
+                                <version>${project.version}</version>
+                                <relativePath/>
+                            </parent>
+                            <bomGroupId>${bomGroupId}</bomGroupId>
+                            <bomArtifactId>${bomArtifactIdPrefix}-ee-preview-with-tools</bomArtifactId>
+                            <bomVersion>${project.version}</bomVersion>
+                            <bomName>${full.dist.product.release.name} BOMs: EE Preview with Tools</bomName>
+                            <bomDescription>Dependency Management for ${full.dist.product.release.name} EE Preview, with Deployment and Testing Tools.</bomDescription>
```
Comment (bstansberry) [inline boms/user/server/ee/ee-preview/ee-preview-with-tools/pom.xml:89]: 'EE' doesn't belong in this text or in line 88.

WildFly Preview is a general purpose preview, not just for new things in Jakarta EE.

File: boms/user/server/expansion/pom.xml [hunk 3]
```diff
@@ -16,77 +16,38 @@
   ~ See the License for the specific language governing permissions and
   ~ limitations under the License.
   -->
```
Comment (bstansberry) [inline boms/user/server/expansion/pom.xml:18]: This should become the standard header.

**Comment by bstansberry** [comment]
Hmm; I thought I was looking at #17990 and made a bunch of comments. So don't think about those too much; I'll look at the right one now.

**Pull request closed without merging**